<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from bioloid_without_legs.xacro     | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="bioloid_without_legs" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!-- Torso -->
  <link name="torso_link">
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/BPF-CHEST-F_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="1.57079632679 0 0"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/BPF-CHEST-B_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="1.57079632679 0 0"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/CM-5_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 3.14159265359 0" xyz="0 0.032 -0.015"/>
    </visual>
    <!-- Sensor 100 -->
    <!--
        <visual>
            <xacro:default_geometry cad_file="AX-S1"/>
            <origin xyz="0 0 0.025" rpy="${-pi/2} 0 ${pi}"/>
        </visual>
        -->
    <!-- Motor 1 -->
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/AX-12_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="1.57079632679 0 -1.57079632679" xyz="-0.032 0 0"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/FP04-F3_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="3.14159265359 -1.57079632679 0" xyz="-0.032 0.019 -0.0145"/>
    </visual>
    <!-- Motor 2 -->
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/AX-12_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="1.57079632679 0 1.57079632679" xyz="0.032 0 0"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/FP04-F3_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="3.14159265359 -1.57079632679 0" xyz="0.032 0.019 -0.0145"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/FP04-F8_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="-1.57079632679 0 1.57079632679" xyz="0 0.0255 -0.0465"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/FP04-F8_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="1.57079632679 0 1.57079632679" xyz="0 0.0255 -0.0843"/>
    </visual>
    <!-- Motor 7 -->
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/AX-12_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 3.14159265359 0" xyz="-0.033 0.016 -0.0655"/>
    </visual>
    <!-- Motor 8 -->
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/AX-12_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 3.14159265359 0" xyz="0.033 0.016 -0.0655"/>
    </visual>
    <collision name="head_collision">
      <origin xyz="0 0.0122 0.035"/>
      <geometry>
        <box size="0.033 0.054 0.044"/>
      </geometry>
    </collision>
    <collision name="torso_collision_top">
      <origin xyz="0 0.019 -0.015"/>
      <geometry>
        <box size="0.103 0.0905 0.0635"/>
      </geometry>
    </collision>
    <collision name="torso_collision_bottom">
      <origin xyz="0 0.00625 -0.0648"/>
      <geometry>
        <box size="0.099 0.065 0.043"/>
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0"/>
      <mass value="1.0"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <!-- 2) Custom parts of modified Bioloid Comprehensive humanoid, with CM-5 replaced by PC and custom sensors -->
  <!--    When using this URDF, ensure a transform from odom to imu_link is being published -->
  <!--
     <xacro:include filename="$(find bioloid_master)/urdf/bioloid_custom_parts.xacro"/> 
-->
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <!--
            <robotNamespace>/bioloid_without_legs</robotNamespace>
-->
      <robotNamespace>/</robotNamespace>
      <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
    </plugin>
  </gazebo>
  <transmission name="right_shoulder_swing_joint_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="right_shoulder_swing_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="right_shoulder_swing_joint_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="left_shoulder_swing_joint_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="left_shoulder_swing_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="left_shoulder_swing_joint_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="right_shoulder_lateral_joint_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="right_shoulder_lateral_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="right_shoulder_lateral_joint_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="left_shoulder_lateral_joint_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="left_shoulder_lateral_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="left_shoulder_lateral_joint_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="right_elbow_joint_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="right_elbow_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="right_elbow_joint_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="left_elbow_joint_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="left_elbow_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="left_elbow_joint_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="right_elbow_joint_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="right_elbow_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="right_elbow_joint_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="left_elbow_joint_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="left_elbow_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="left_elbow_joint_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- World -->
  <!-- Used for fixing robot to Gazebo 'base_link' -->
  <!--
    <link name="world"/>

    <joint name="world_to_odom" type="fixed">
        <parent link="world"/>
        <child link="odom"/>
        <origin xyz="0 0 0.3" rpy="0 0 0"/>
      </joint>

    <link name="odom"/>

      <joint name="odom_to_imu" type="fixed">
        <parent link="odom"/>
        <child link="imu"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
      </joint>
-->
  <link name="imu"/>
  <joint name="imu_to_base_link" type="fixed">
    <parent link="imu"/>
    <child link="base_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <!--
    <joint name="imu_to_base" type="fixed">
        <parent link="imu"/>
        <child link="base_link"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
      </joint>
-->
  <!-- Base -->
  <joint name="base_to_torso_joint" type="fixed">
    <parent link="base_link"/>
    <origin rpy="0 0 1.57079632679" xyz="0 0 0"/>
    <child link="torso_link"/>
  </joint>
  <link name="base_link">
    </link>
  <!-- Neck Joint-->
  <joint name="head_yaw_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.015"/>
    <parent link="base_link"/>
    <child link="neck_link"/>
    <limit effort="10" lower="-3.14159265359" upper="3.14159265359" velocity="1.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="0" velocity="0"/>
  </joint>
  <!-- Neck Link-->
  <link name="neck_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://bioloid_example/mesh/neck_link.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
  </link>
  <!-- Head Link-->
  <link name="head_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://bioloid_example/mesh/head_link.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.5 0.5 0.5 1"/>
      </material>
    </visual>
  </link>
  <!-- Head Joint-->
  <joint name="head_pitch_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.008 0 0.053"/>
    <parent link="neck_link"/>
    <child link="head_link"/>
    <limit effort="10" lower="-1.3" upper="1.4" velocity="1.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="0" velocity="0"/>
  </joint>
  <joint name="right_shoulder_swing_joint" type="revolute">
    <parent link="torso_link"/>
    <!-- <origin xyz="${reflect*(-0.032)} 0 0" rpy="0 ${reflect*(-pi/2)} ${x*pi}"/> -->
    <origin rpy="-1.57079632679 0 1.57079632679" xyz="-0.032 0 0"/>
    <axis xyz="0 0 1"/>
    <child link="right_shoulder_swing_link"/>
    <limit effort="10" lower="-3.14159265359" upper="3.14159265359" velocity="1.0"/>
  </joint>
  <link name="right_shoulder_swing_link">
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/FP04-F1_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 3.14159265359 -1.57079632679" xyz="-0.014 0 0.044"/>
    </visual>
    <collision name="right_shoulder_collision">
      <origin xyz="-0.006 0 0.0375"/>
      <geometry>
        <box size="0.039 0.052 0.036"/>
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0"/>
      <mass value="1.0"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <joint name="left_shoulder_swing_joint" type="revolute">
    <parent link="torso_link"/>
    <!-- <origin xyz="${reflect*(-0.032)} 0 0" rpy="0 ${reflect*(-pi/2)} ${x*pi}"/> -->
    <origin rpy="1.57079632679 0 -1.57079632679" xyz="0.032 0 0"/>
    <axis xyz="0 0 1"/>
    <child link="left_shoulder_swing_link"/>
    <limit effort="10" lower="-3.14159265359" upper="3.14159265359" velocity="1.0"/>
  </joint>
  <link name="left_shoulder_swing_link">
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/FP04-F1_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 1.57079632679" xyz="0.014 0 -0.044"/>
    </visual>
    <collision name="left_shoulder_collision">
      <origin xyz="0.006 0 -0.0375"/>
      <geometry>
        <box size="0.039 0.052 0.036"/>
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0"/>
      <mass value="1.0"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <joint name="right_shoulder_lateral_joint" type="revolute">
    <parent link="right_shoulder_swing_link"/>
    <!-- <origin xyz="${reflect*(-0.014)} 0 ${reflect*0.044}" rpy="${-pi/2} 0 0"/> -->
    <origin rpy="-1.57079632679 1.57079632679 0" xyz="-0.014 0 0.044"/>
    <axis xyz="0 0 1"/>
    <child link="right_upper_arm_link"/>
    <limit effort="10" lower="-3.14159265359" upper="3.14159265359" velocity="1.0"/>
  </joint>
  <link name="right_upper_arm_link">
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/AX-12_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="3.14159265359 0 1.57079632679" xyz="0 0 0"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/BPF-BU_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="1.57079632679 0 0" xyz="0 0 0.0185"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/BPF-WA_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="1.57079632679 0 0" xyz="0 0 0.019"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/FP04-F3_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 -1.57079632679" xyz="-0.0415 0 0"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/FP04-F2_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 1.57079632679 0" xyz="-0.0675 0 0"/>
    </visual>
    <collision name="right_upper arm_collision_proximal">
      <origin xyz="-0.0148 0 0"/>
      <geometry>
        <box size="0.054 0.033 0.044"/>
      </geometry>
    </collision>
    <collision name="right_upper arm_collision_distal">
      <origin xyz="-0.0555 0 0"/>
      <geometry>
        <box size="0.047 0.033 0.052"/>
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0"/>
      <mass value="1.0"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <joint name="left_shoulder_lateral_joint" type="revolute">
    <parent link="left_shoulder_swing_link"/>
    <!-- <origin xyz="${reflect*(-0.014)} 0 ${reflect*0.044}" rpy="${-pi/2} 0 0"/> -->
    <origin rpy="-1.57079632679 1.57079632679 0" xyz="0.014 0 -0.044"/>
    <axis xyz="0 0 1"/>
    <child link="left_upper_arm_link"/>
    <limit effort="10" lower="-3.14159265359" upper="3.14159265359" velocity="1.0"/>
  </joint>
  <link name="left_upper_arm_link">
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/AX-12_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 1.57079632679" xyz="0 0 0"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/BPF-BU_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="-1.57079632679 0 0" xyz="0 0 -0.0185"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/BPF-WA_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="-1.57079632679 0 0" xyz="0 0 -0.019"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/FP04-F3_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 1.57079632679" xyz="0.0415 0 0"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/FP04-F2_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 -1.57079632679 0" xyz="0.0675 0 0"/>
    </visual>
    <collision name="left_upper arm_collision_proximal">
      <origin xyz="0.0148 0 0"/>
      <geometry>
        <box size="0.054 0.033 0.044"/>
      </geometry>
    </collision>
    <collision name="left_upper arm_collision_distal">
      <origin xyz="0.0555 0 0"/>
      <geometry>
        <box size="0.047 0.033 0.052"/>
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0"/>
      <mass value="1.0"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <joint name="right_elbow_joint" type="revolute">
    <parent link="right_upper_arm_link"/>
    <origin rpy="0 0 0" xyz="-0.0675 0 0"/>
    <axis xyz="0 0 1"/>
    <child link="right_lower_arm_link"/>
    <limit effort="10" lower="-1.77" upper="1.77" velocity="1.0"/>
  </joint>
  <link name="right_lower_arm_link">
    <!-- Motor 5 -->
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/AX-12_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="3.14159265359 0 1.57079632679" xyz="0 0 0"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/BPF-BU_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="1.57079632679 0 0" xyz="0 0 0.0185"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/BPF-WA_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="1.57079632679 0 0" xyz="0 0 0.019"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/FP04-F3_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 -1.57079632679" xyz="-0.0415 0 0"/>
    </visual>
    <collision name="right_lower_arm_collision">
      <origin xyz="-0.0135 0 0"/>
      <geometry>
        <box size="0.051 0.033 0.044"/>
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0"/>
      <mass value="1.0"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <joint name="left_elbow_joint" type="revolute">
    <parent link="left_upper_arm_link"/>
    <origin rpy="0 0 0" xyz="0.0675 0 0"/>
    <axis xyz="0 0 1"/>
    <child link="left_lower_arm_link"/>
    <limit effort="10" lower="-1.77" upper="1.77" velocity="1.0"/>
  </joint>
  <link name="left_lower_arm_link">
    <!-- Motor 6 -->
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/AX-12_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 1.57079632679" xyz="0 0 0"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/BPF-BU_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="-1.57079632679 0 0" xyz="0 0 -0.0185"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/BPF-WA_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="-1.57079632679 0 0" xyz="0 0 -0.019"/>
    </visual>
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/FP04-F3_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 1.57079632679" xyz="0.0415 0 0"/>
    </visual>
    <collision name="left_lower_arm_collision">
      <origin xyz="0.017 0 0"/>
      <geometry>
        <box size="0.058 0.033 0.044"/>
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0"/>
      <mass value="1.0"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <joint name="right_hand_joint" type="fixed">
    <parent link="right_lower_arm_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <child link="right_hand_link"/>
  </joint>
  <link name="right_hand_link">
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/FP04-F11_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="-3.14159265359 1.57079632679 0" xyz="-0.0415 0 0"/>
    </visual>
    <collision name="right_palm_collision">
      <origin xyz="-0.039 0 0"/>
      <geometry>
        <box size="0.0131 0.0265 0.044"/>
      </geometry>
    </collision>
    <collision name="right_fingers_collision">
      <origin xyz="-0.0701 -0.0077 0"/>
      <geometry>
        <box size="0.0575 0.0175 0.035"/>
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0"/>
      <mass value="1.0"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <joint name="left_hand_joint" type="fixed">
    <parent link="left_lower_arm_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <child link="left_hand_link"/>
  </joint>
  <link name="left_hand_link">
    <visual>
      <geometry>
        <mesh filename="file:///home/mateus/catkin_ws/src/ros-bioloid-master/src/bioloid_master/../../assets/cad/FP04-F11_FCD-Conv_BDR-Col.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 1.57079632679 0" xyz="0.0415 0 0"/>
    </visual>
    <collision name="left_palm_collision">
      <origin xyz="0.039 0 0"/>
      <geometry>
        <box size="0.0131 0.0265 0.044"/>
      </geometry>
    </collision>
    <collision name="left_fingers_collision">
      <origin xyz="0.0701 0.0077 0"/>
      <geometry>
        <box size="0.0575 0.0175 0.035"/>
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0"/>
      <mass value="1.0"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
</robot>

